examples/parameters/main.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	dbsql "github.com/databricks/databricks-sql-go"
+	"github.com/joho/godotenv"
+)
+
+func main() {
+	// Opening a driver typically will not attempt to connect to the database.
+	err := godotenv.Load()
+
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	port, err := strconv.Atoi(os.Getenv("DATABRICKS_PORT"))
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	connector, err := dbsql.NewConnector(
+		dbsql.WithServerHostname(os.Getenv("DATABRICKS_HOST")),
+		dbsql.WithPort(port),
+		dbsql.WithHTTPPath(os.Getenv("DATABRICKS_HTTPPATH")),
+		dbsql.WithAccessToken(os.Getenv("DATABRICKS_ACCESSTOKEN")),
+	)
+	if err != nil {
+		// This will not be a connection error, but a DSN parse error or
+		// another initialization error.
+		log.Fatal(err)
+	}
+	db := sql.OpenDB(connector)
+	defer db.Close()
+	// ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	// defer cancel()
+	ctx := context.Background()
+	var p_bool bool
+	var p_int int
+	var p_double float64
+	var p_float float32
+	var p_date string
+	err1 := db.QueryRowContext(ctx, `SELECT
+	:p_bool AS col_bool,
+	:p_int AS col_int,
+	:p_double AS col_double,
+	:p_float AS col_float,
+	:p_date AS col_date`,
+		dbsql.DBSqlParam{Name: "p_bool", Value: true},
```

Comment:
nothing to do with this PR, but why is this called DBSqlParam? Please, please standardize the names we expose as public interfaces. @rcypher-databricks what do you think?

Reply:
we should just call this Param. As it is already scoped to dbsql

Reply:
@nithinkdb I though we talked about dropping DBSql and just calling it Parameter.

Reply:
Sorry, forgot about this. Will do.